core/src/main/scala/org/ensime/core/Refactoring.scala

```diff
@@ -24,7 +25,16 @@ abstract class RefactoringEnvironment(file: String, start: Int, end: Int) {
     procId: Int,
     tpe: RefactorType,
     parameters: refactoring.RefactoringParameters
-  ): Either[RefactorFailure, RefactorEffect] = {
+  )(implicit charset: Charset): Either[RefactorFailure, RefactorDiffEffect] = {
```

Comment:
the charset is used when reading the user's files, but we can choose to use utf-8 for our diff file.